geomstats/datasets/prepare_emg_data.py

```diff
@@ -0,0 +1,116 @@
+"""Pre-process emg time series into batched covariance matrices.
+
+The user defines the number of time steps of the batches.
+"""
+
+import numpy as np
+import pandas as pd
+
+
+def _vectorize_cov(cov_mat):
+    """Vectorize a symmetric Matrix.
+
+    Convert a symetric matrice of size n x n into a vector containing
+    the n(n+1) lower elements (diagonal included).
+
+    Parameters
+    ----------
+    cov_mat : np.array
+        covariance matrix to vectorize.
+
+    Returns
+    -------
+    covec : np.array
+        vector of the n(n+1) lower elements of the matrix.
+    """
+    lcov = len(cov_mat)
+    covec = []
+    for i in range(lcov):
+        for j in range(i + 1):
+            covec.append(cov_mat[i][j])
+    return np.array(covec)
+
+
+class TimeSerieCovariance:
+    """Class for generating a list of covariance matrix from time series.
+
+    It starts by removing the transient signal by taking a margin on each side
```

Comment:
This documentation, and some parts of the code, seem to be specific to your EMG data (since it incorporates sign changes?). Two options:
- Either only keep general code in this class, that could be applied to any time-series. The code specific to your EMG data would go to the ipynb.
- Or make it clear that the preprocessing is meant for your EMG data, and give the link to the data in the docstring.

I would maybe prefer option 1, because it is likely that such general code (creation of batches and covariance matrices from any timeseries) will be interesting to users. But this is your choice!